Services/Init/classes/class.ilStartUpGUI.php

```diff
@@ -271,6 +278,24 @@ public static function printToGlobalTemplate($tpl)
         $gtpl->setContent($tpl->get());
         $gtpl->printToStdout("DEFAULT", false, true);
     }
+    
+    protected function retrieveMessagesFromSession() : array
+    {
+        $messages = [];
+        if (isset($_SESSION[ilGlobalTemplate::MESSAGE_TYPE_FAILURE])) {
+            $messages[ilGlobalTemplate::MESSAGE_TYPE_FAILURE] = $_SESSION[ilGlobalTemplate::MESSAGE_TYPE_FAILURE];
```

Comment:
Please avoid the direct access to $_SESISON and use ilSession.

Reply:
Done